beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -144,3 +149,18 @@ func (r *Service) setSeenBlockIndexSlot(slot uint64, proposerIdx uint64) {
 	b := append(bytesutil.Bytes32(slot), bytesutil.Bytes32(proposerIdx)...)
 	r.seenBlockCache.Add(string(b), true)
 }
+
+// This captures metrics for block arrival time by subtracts slot start time.
+func captureArrivalTimeMetric(genesisTime uint64, currentSlot uint64) error {
+	startTime, err := helpers.SlotToTime(genesisTime, currentSlot)
+	if err != nil {
+		return err
+	}
+	diff := uint64(roughtime.Now().UnixNano() - startTime.UnixNano())
+
+	// Denominate everything in milliseconds.
```

Comment:
This would actually be microseconds. If you use the time API as suggested above, it’s more clear

1 microsecond is 1000ns.